services/src/main/scala/cromwell/services/healthmonitor/HealthMonitorServiceActor.scala

```diff
@@ -32,49 +34,71 @@ trait HealthMonitorServiceActor extends Actor with LazyLogging with Timers {
 
+  lazy val sweepInterval = serviceConfig.as[Option[FiniteDuration]]("services.HealthMonitor.check-refresh-time").getOrElse(DefaultSweepTime)
   lazy val futureTimeout: FiniteDuration = serviceConfig.as[Option[FiniteDuration]]("services.HealthMonitor.check-timeout").getOrElse(DefaultFutureTimeout)
   lazy val staleThreshold: FiniteDuration = serviceConfig.as[Option[FiniteDuration]]("services.HealthMonitor.status-ttl").getOrElse(DefaultStaleThreshold)
-
-  override def preStart(): Unit = {
-    val sweepTime = serviceConfig.as[Option[FiniteDuration]]("services.HealthMonitor.check-refresh-time").getOrElse(DefaultSweepTime)
-    timers.startPeriodicTimer(CheckTickKey, CheckAll, sweepTime)
-    logger.info("Starting health monitor with the following checks: " + subsystems.map(_.name).mkString(", "))
-  }
+  lazy val failureRetryCount: Int = serviceConfig.as[Option[Int]]("services.HealthMonitor.check-failure-retry-count").getOrElse(DefaultFailureRetryCount)
+  lazy val failureRetryInterval: FiniteDuration = serviceConfig.as[Option[FiniteDuration]]("services.HealthMonitor.check-failure-retry-interval").getOrElse(DefaultFailureRetryInterval)
 
   /**
     * Contains each subsystem status along with a timestamp of when the entry was made so we know when the status
     * goes stale. Initialized with unknown status.
     */
-  private var statusCache: Map[MonitoredSubsystem, CachedSubsystemStatus] = {
+  private[healthmonitor] var statusCache: Map[MonitoredSubsystem, CachedSubsystemStatus] = {
     val now = System.currentTimeMillis
     subsystems.map((_, CachedSubsystemStatus(UnknownStatus, now))).toMap
   }
 
+  subsystems foreach { s => self ! Check(s, failureRetryCount) }
+
+  private def check(subsystem: MonitoredSubsystem, after: FiniteDuration, withRetriesLeft: Int): Unit = {
+    timers.startSingleTimer(UUID.randomUUID(), Check(subsystem, withRetriesLeft), after)
+  }
+
+  private def scheduleFailedRetryCheck(subsystem: MonitoredSubsystem, retriesLeft: Int): Unit = {
+    check(subsystem, after = failureRetryInterval, withRetriesLeft = retriesLeft - 1)
+  }
+
+  private[healthmonitor] def scheduleSweepCheck(subsystem: MonitoredSubsystem): Unit = {
+    check(subsystem, after = sweepInterval, withRetriesLeft = failureRetryCount)
+  }
+
   override def receive: Receive = {
-    case CheckAll =>
-      logger.debug("Checking status on all subsystems")
-      subsystems.foreach(checkSubsystem)
-    case Store(subsystem, status) => store(subsystem, status)
+    case Check(subsystem, retriesLeft) =>
+      checkSubsystem(subsystem, retriesLeft)
+    case Store(subsystem, status) =>
+      store(subsystem, status)
+      scheduleSweepCheck(subsystem)
     case GetCurrentStatus => sender ! getCurrentStatus
     case ShutdownCommand => context.stop(self) // Not necessary but service registry requires it. See #2575
     case Status.Failure(f) => throw f
-    case _ => throw new RuntimeException("wtf")
+    case e => throw new RuntimeException(e.toString)
```

Comment:
ToL: is this really wanted ? Any unrecognized message should kill the actor ?

Reply:
Good question but preexisting, I only made the message possibly less useless.

Reply:
Okay, I'd lean towards logging an error rather than crashing the whole thing but 🤷‍♂️